### PR TITLE
Refactor shared types 

### PR DESCRIPTION
**Pull request by belgamo**
I noticed that by trying to reuse a single type definition for everything (dto, db entities, responses) we were falling into a trap. It's better to have separate types (although it feels like we're duplicating) because they represent and mean different things.

Remember that our final goal in having shared interfaces is to be able to guarantee the contract between the front and backend.

### CODE REVIEW

File: apps/backend/package.json [hunk 1]
```diff
@@ -14,14 +14,14 @@
...
-    "test": "jest",
-    "test:watch": "jest --watch",
-    "test:cov": "jest --coverage",
-    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json"
+    "test": "TZ=UTC jest",
+    "test:watch": "TZ=UTC jest --watch",
+    "test:cov": "TZ=UTC jest --coverage",
+    "test:debug": "TZ=UTC node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "TZ=UTC jest --config ./test/jest-e2e.json"
```
Comment (brunohkbx) [inline apps/backend/package.json]: I wonder if wouldn't be better to have this env declared in the jest configuration section at this file

File: apps/backend/src/user/repositories/user-repository.interface.ts [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
-import { IUser } from './mongodb/user.interface';
+import { Exclude } from 'class-transformer';
+
+export class UserEntity {
+  id: string;
+  name: string;
+  email: string;
+
+  @Exclude()
+  password: string;
```
Comment (brunohkbx) [inline apps/backend/src/user/repositories/user-repository.interface.ts:9]: Is there any alternative way of doing the same without needing a new package? I wonder if Omit works here. Found this example: https://www.prisma.io/docs/concepts/components/prisma-client/excluding-fields#excluding-the-password-field

File: apps/backend/src/adoption/repositories/adoption-repository.interface.ts [hunk 3]
```diff
@@ -1,23 +1,50 @@
...
-export type IAdoption = AdoptionType & {
-  user: string;
-  location: Location;
+export class AdoptionEntity {
+  id: string;
+  name: string;
+  gender: AnimalGendersType;
+  breed: string;
+  type: AnimalType;
+  age: AnimalAgesType;
```
Comment (brunohkbx) [inline apps/backend/src/adoption/repositories/adoption-repository.interface.ts:15]: Out of the scope of this PR but I feel like `age` sounds confusing here as in implying to be a numeric value. I wonder if there's a better name to represent the values of this enum

File: apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts [hunk 4]
```diff
@@ -16,39 +19,46 @@ export class AdoptionMongoDBRepository implements AdoptionRepository {
...
-    const newAdoption = new this.adoptionModel(adoption);
+  async create(adoption: AdoptionEntity) {
+    const newAdoption = new this.adoptionModel(AdoptionMap.toSchema(adoption));
 
     const document = await newAdoption
       .save()
-      .then((adoption) => adoption.populate('user', 'id name'));
+      .then((adoption) =>
+        adoption.populate<PopulatedAdoptionDocument>('user', 'id name'),
+      );
```
Comment (brunohkbx) [inline apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts]: Does it work without this `then` call? just noticed that the function below doesn't have it

```suggestion
      .populate<PopulatedAdoptionDocument>('user', 'id name')
```

File: apps/backend/test/e2e/adoptions.e2e-spec.ts [hunk 5]
```diff
@@ -21,6 +21,8 @@ import { AuthService } from '../../src/auth/auth.service';
 import { ConfigModule } from '@nestjs/config';
 import { getCoordinatesFromUser } from '../../src/user/user.helpers';
 
+Date.now = () => 1000;
```
Comment (brunohkbx) [inline apps/backend/test/e2e/adoptions.e2e-spec.ts]: Is this change intentional?

File: apps/backend/test/mocks/user.ts [hunk 6]
```diff
@@ -1,6 +1,20 @@
...
+  password: 'quesenharuimgrosbilda',
+  location: {
+    latitude: 10.002322,
+    longitude: -40.343569,
+  },
+  photoUri: 'http://www.google.com/',
+  id: '123',
+  createdAt: '',
+  updatedAt: '',
+};
```
Comment (brunohkbx) [inline apps/backend/test/mocks/user.ts:15]: Also out of the scope of this PR but would be nice to set up [Faker](https://fakerjs.dev/) for that

Comment (belgamo) [reply apps/backend/test/mocks/user.ts:15]: Thanks, addressing an issue for that https://github.com/animavita/animavita/issues/185

File: apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts (hunk 4 above)
Comment (belgamo) [reply apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts]: Yep, it does. Good catch!

File: apps/backend/src/adoption/repositories/adoption-repository.interface.ts (hunk 3 above)
Comment (belgamo) [reply apps/backend/src/adoption/repositories/adoption-repository.interface.ts:15]: It used to be a number. However, we decided to come up with a generic way of representing the age of the pet since it might be a dog, a cat, or any other animal. Right now I can't think of a better name for that, but it makes sense. Addressing an issue for that https://github.com/animavita/animavita/issues/186

File: apps/backend/package.json (hunk 1 above)
Comment (belgamo) [reply apps/backend/package.json]: For sure, thanks!

File: apps/backend/src/user/repositories/user-repository.interface.ts (hunk 2 above)
Comment (belgamo) [reply apps/backend/src/user/repositories/user-repository.interface.ts:9]: The advantage of using this approach is that Nest automatically excludes the desired properties through interceptors https://docs.nestjs.com/techniques/serialization#exclude-properties

File: apps/backend/test/e2e/adoptions.e2e-spec.ts (hunk 5 above)
Comment (belgamo) [reply apps/backend/test/e2e/adoptions.e2e-spec.ts]: It was a quick test and I forgot to remove. Thanks
